File: src/main/kotlin/akuchars/domain/task/model/ChangeEntityTime.java
package akuchars.domain.task.model;

import kotlin.jvm.internal.Intrinsics;

import java.time.LocalDateTime;

class ChangeEntityTime {
	private LocalDateTime createdDate;
	private LocalDateTime updateDate;

	private ChangeEntityTime(LocalDateTime createdDate, LocalDateTime updateDate) {
		Intrinsics.checkParameterIsNotNull(createdDate, "createdDate");
		this.createdDate = createdDate;
		this.updateDate = updateDate;
	}

	ChangeEntityTime() {
	}

	static ChangeEntityTime now() {
		return new ChangeEntityTime(LocalDateTime.now(), null);
	}

	void updateTime() {
		this.updateDate = LocalDateTime.now();
	}
}

File: src/main/kotlin/akuchars/domain/task/model/ChangeEntityTime.kt
package akuchars.domain.task.model

import java.time.LocalDateTime
import javax.persistence.Embeddable

@Embeddable
data class ChangeEntityTime(
		val createdDate: LocalDateTime,
		private var updateDate: LocalDateTime?
) {

	internal fun updateTime() {
		this.updateDate = LocalDateTime.now()
	}

	companion object {

		@JvmStatic
		internal fun now(): ChangeEntityTime {
			return ChangeEntityTime(LocalDateTime.now(), null)
		}
	}
}
